File: testdata/FizzBuzz.java
public class FizzBuzz {
    public int parseMax(String[] args) {
        if (args.length == 0) {
            return 100;
        }
        try {
            return Integer.parseInt(args[0]);
        } catch (NumberFormatException e) {
            return 100;
        }
    }

    public String fizzbuzz(int n) {
        if (n % 15 == 0) {
            return "FizzBuzz";
        } else if (n % 3 == 0) {
            return "Fizz";
        } else if (n % 5 == 0) {
            return "Buzz";
        } else {
            return Integer.toString(n);
        }
    }

    public void run(String[] args) {
        int max = parseMax(args);
        for(int i = 1; i <= max; i++) {
            System.out.printf("%d: %s%n", i, fizzbuzz(i));
        }
    }

    public static void main(String[] args) {
        new FizzBuzz().run(args);
    }
}

File: testdata/fizzbuzz.rs
fn parse_args(args: Vec<String>) -> u32 {
    if args.len() < 2 {
        100
    } else {
        args[1].parse().unwrap_or(100)
    }
}

fn fizzbuzz(i: u32) -> String {
    match (i % 3, i % 5) {
        (0, 0) => "FizzBuzz".to_string(),
        (0, _) => "Fizz".to_string(),
        (_, 0) => "Buzz".to_string(),
        _ => i.to_string(),
    }
}

fn perform(max: u32) {
    for i in 1..=max {
        println!("{i}, {}", fizzbuzz(i));
    }
}

fn main() {
    let args = std::env::args().collect::<Vec<String>>();
    let max = parse_args(args);
    perform(max);
}